### DISCUSSION TITLE
[Feature Request]: Ability to physically connect multiple nodes together to share messages as if they were one single node

### DISCUSSION THREAD

**Discussion by Bcpratt2000** [meshtastic/firmware#4841] [Ideas]
### Platform

Cross-Platform

### Description

**Request**
I would like the ability to connect multiple Lora boards **Physically** over I2C or Serial, and share messages between them as if they were a single node. I would bluetooth connect to one of the nodes and it would sync messages/position reports/etc. between N number of radios. Additionally, I would like the option of cross-node repeating If board 1 recieves a packet, board 2 can repeat that packet(See use case 2).

**Use Case:**
1. I have a tracker node that I attach to my dog's collar when backpacking/outside. My ideal modem preset for that node is "Very Long Slow" because it is only for very low bandwidth, long range position reports, often in the backcountry without much infrastructure. In order to use this, I would need to set my pocket node to "Very Long Slow" However, I would also like to be able to monitor the local LongFast/MediumFast Traffic (3 modem presets total, for 3 different nodes). In this case I would not like to repeat messages cross-node.
2. If I am wanting to bring coverage to a local area that is far away from a Router, I can connect a node with a high-gain yagi antenna physically to a node with a dipole (for local coverage) and treat that as a single Router node. This would ensure a solid connection to the rest of the network, without compromising local area connectivity.

**Comment by tedwardd** [comment]
The ability to connect two meshes together over a point-to-point link **without encurring a hop penalty or getting tagged as from an MQTT source** would be an absolute boon for meshastic coverage. I've considered this for a setup to link two remote communities together but as it stands right now, the only theoretical approach is to link two nodes via MQTT. This, however, both incurs a hop penalty on each side of the link as well as tagging the messages as coming from MQTT which is very often filtered out by nodes within the mesh.

In order to solve both these issues, you would need the ability to pass a message out of band, via I2C, serial or other physical means so that a message received by an omnidirectional node (the node that is "part of the local mesh" if you will) can transmit from the node with the directional antenna (the remote link). The second piece, avoiding hop penalty, would mean that the directional links on either side would need to pass the message transparently as well. You could accomplish this using the same DH encryption model used by remote admin now.

Each end of the long-distance link would be configured with the public key of the other. The sending node then encapsulates the original packet within a new, encrypted message (or with a signed header or other mechanism to ensure valid source and message integrity) to the other. Doing this removes the possibility of abuse of the "skipped hop" by outside actors, ensuring the hop-count integrity of the original message is maintained across the link.

**Reply by ellisgl** [reply]
I see a use case of having directional antennas and repeating out other antennas, instead of repeating it back the direction it came.

**Reply by GUVWAF** [reply]
I would not recommend using multiple directional antennas, and maybe in general no directional antennas. 
First, the benefit of directional antennas is limited; line-of-sight is key, it doesn't really matter if originally your power is higher if further on the signal needs to get through a mountain. Also you need to check with your local regulations if more than 0dBi gain is even allowed as Meshtastic sets the transmission power by default to the maximum of your region.
Secondly, transmitting in only one direction can cause very weird behavior on the mesh. Nodes actually listen if someone repeats their message, otherwise they will retransmit, so if they don't hear that, they will keep on retransmitting. Also, nodes need to apply listen-before-talk to ensure they're not colliding with other nodes. If that is asymmetric, the chance of collisions because of the hidden node problem gets higher.

**Reply by ellisgl** [reply]
That makes sense. Didn't think about ACKing.

**Reply by n0p** [reply]
I think having a repeater on a good position with sector antennas, each one pointing to a zone is a valid use case.

**Comment by KamikazeYasha** [comment]
A great feature, I have a few projects in which it would be used.
